src/Savings.java

```diff
@@ -1,2 +1,87 @@
-public class Savings {
+/**
+ * Savings Class extending Account Class
+ * @author Seifeldeen Mohamed
+ */
+public class Savings extends Account {
+    protected boolean isLoyal;
+    // Constants for interest rate and fee
+    private static final double DEFAULT_INTEREST_RATE = 0.02; // 2% interest rate
+    private static final double LOYALTY_INTEREST_RATE = 0.0425; // 4.25% interest rate for loyal customers
+    private static final double MONTHLY_FEE = 25.00;
+    private static final double MINIMUM_BALANCE_FOR_NO_FEE = 500.00;
+    private static final double NO_FEE = 0.0;
+
+    /**
+     * Contrustor with default values
+     *
+     * @param profileHolder
+     * @param balance
+     * @param isLoyal
+     */
+    public Savings(Profile profileHolder, double balance, boolean isLoyal) {
+        super(profileHolder, balance);
+        this.isLoyal = isLoyal;
+    }
+
+    /**
+     * boolean Function to check if the customer isLoyal or not
+     * @return
+     */
+    public boolean isLoyal() {
+        return isLoyal;
+    }
+
+    /**
+     * method to caluclate the monthly interest
+     * @return
+     */
+    @Override
+    public double monthlyInterest() {
+        // if the customer is loyal, then they get the loyal interest rate otherwise give them the default
+        double interestRate = isLoyal ? LOYALTY_INTEREST_RATE : DEFAULT_INTEREST_RATE;
+        return balance * interestRate / 12;
+    }
+
+    /**
+     *
+     * @return
+     */
+    @Override
+    public double monthlyFee() {
+        if(balance <= MINIMUM_BALANCE_FOR_NO_FEE){
+            return NO_FEE;
+        }
+        return MONTHLY_FEE;
+    }
+
+    @Override
+    public int compareTo(Account OtherAccount) {
+        if (OtherAccount instanceof Savings otherSavingsAccount) {
+            // Compare Account based on the balance
+            return Double.compare(this.balance, otherSavingsAccount.balance);
```

Comment:
I don't see why we only compare savings accounts by account balance and not just all accounts. Are money market funds and college savings accounts also compared differently? This seems like it belongs in the superclass, it does not use any savings account specific methods/instance variables.